src/test/java/jp/co/soramitsu/sora/util/SignatureGenerator.java

```diff
@@ -1,51 +1,78 @@
-import jp.co.soramitsu.sora.crypto.DocumentSignatureService.NoSuchStrategy;
-import jp.co.soramitsu.sora.crypto.algorithms.RawSignatureStrategy.SignatureSuiteException;
-import jp.co.soramitsu.sora.crypto.hash.Sha3Digest256;
+import jp.co.soramitsu.sora.crypto.common.SecurityProvider;
+import jp.co.soramitsu.sora.crypto.json.JSONCanonizerWithOneCoding;
+import jp.co.soramitsu.sora.crypto.proof.Options;
+import jp.co.soramitsu.sora.crypto.signature.suite.JSONEd25519Sha3SignatureSuite;
+import jp.co.soramitsu.sora.crypto.type.SignatureTypeEnum;
 import jp.co.soramitsu.sora.didresolver.dto.DDO;
+import jp.co.soramitsu.sora.didresolver.dto.Proof;
+import lombok.val;
 
 public class SignatureGenerator {
 
   private static final String PATH = "src/test/resources/";
 
   public static void main(String[] args)
-      throws IOException, SignatureSuiteException, NoSuchStrategy, CryptoException {
-    DocumentSignatureService documentSignatureService = new DocumentSignatureService(
-        new Sha3Digest256());
-    ObjectMapper objectMapper = new ObjectMapper();
-    objectMapper.registerModule(new JavaTimeModule());
-    objectMapper.setSerializationInclusion(Include.NON_NULL);
-    Reader jsonReader = new BufferedReader(
-        new InputStreamReader(
-            SignatureGenerator.class.getClassLoader().getResourceAsStream("ddo.json")));
-    DDO ddo = objectMapper.readValue(jsonReader, DDO.class);
-    KeyPair keyPair = new Ed25519Sha3().generateKeypair();
+      throws IOException, SignatureException, NoSuchAlgorithmException {
+
+    ObjectMapper objectMapper = new ObjectMapper()
+        .registerModule(new JavaTimeModule())
+        .enable(INDENT_OUTPUT)
+        .setSerializationInclusion(NON_NULL);
+
+    JSONEd25519Sha3SignatureSuite documentSignatureService = new JSONEd25519Sha3SignatureSuite(
+        new SecurityProvider(), new JSONCanonizerWithOneCoding(), objectMapper);
+
+    InputStream in = SignatureGenerator.class.getClassLoader().getResourceAsStream("ddo.json");
+    DDO ddo = objectMapper.readValue(in, DDO.class);
+
+    KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance(KEY_ALGORITHM);
+    KeyPair keyPair = keyPairGenerator.generateKeyPair();
     writeKeyPairToFile(keyPair);
-    ddo.getPublicKey().get(1).setPublicKeyValue(((EdDSAPublicKey) keyPair.getPublic()).getAbyte());
-    documentSignatureService.sign(ddo, keyPair, ddo.getProof());
+
+    ddo.getPublicKey()
+        .get(0)
+        .setPublicKeyValue(keyPair.getPublic().getEncoded());
+
+    Proof proof = ddo.getProof();
+    Options options = new Options(
+        SignatureTypeEnum.valueOf(proof.getType()),
+        proof.getCreated().toString(),
+        proof.getCreator().toString(),
+        proof.getNonce(),
+        proof.getPurpose()
+    );
+
+    val signed = documentSignatureService
+        .sign(ddo, (EdDSAPrivateKey) keyPair.getPrivate(), options);
+
+    assert documentSignatureService.verify(signed, (EdDSAPublicKey) keyPair.getPublic());
```

Comment:
Also, don't rely on this, assertions IIRC can be simply switched off at JVM startup

Reply:
I know. This assert is just self-check.